ml-proto/spec/eval.ml

```diff
@@ -88,6 +88,9 @@ let numerics_error at = function
       error at "runtime: invalid conversion to integer"
   | exn -> raise exn
 
+let some_memory c =
+  match c.module_.memory with Some m -> m | _ -> assert false
```

Comment:
That sounds good.  So maybe a new `Soundness` exn type?  Since this shouldn't ever happen, seems like it could be made more terse to raise and leave out position/string.